release/models/platform/openconfig-platform-pipeline-counters.yang

```diff
@@ -1181,6 +1188,130 @@ module openconfig-platform-pipeline-counters {
     }
   }
 
+  grouping control-plane-traffic-counters-state {
```

Comment:
Do we believe that this is the maximum commonality of these counters that we can get? It seems to me like there is an opportunity for us to have breakdowns of counters based on the configured control-plane protection/classifiers that are there.... would this be handled in the proposed way to add such policers? If not, is there a proposal as to how to handle this?

Reply:
I can speculate that JunOS may have flexibility report counters based on the configured policy as they appear to use the same firewall filters used on regular interfaces.  But the other implementations appear to have configuration and counters that are unique to control plane traffic and not shared with QoS and ACL counters.  

@rolandphung @aredmon8551 @rgwilton @hobbydos for comment on their platform's ability to to report control plane protection counters using defined QoS and ACL rules vs. system defined/fixed counters specific to control plane protection.

I think we should keep front of mind the operational use cases for these counters which are the same as the pipeline counters defined in #715 to which these are an expansion of the same pattern.  The use cases being:
1.  coarse identification of the cause or source of device packet loss as either intended (due to policy) or unintended (due to device health/error) to make a decision on isolating/draining the device to mitigate loss due if the cause is device errors/un-healthy device. 
2. After mitigation, assist deep root cause analysis by making time series debug data available to the device designer/owner which includes device hardware specific counters.

Reply:
In #762 (Control Plane Protection model) the referenced QoS and [defined sets ACL](https://github.com/openconfig/public/blob/master/release/models/defined-sets/openconfig-defined-sets.yang) models have counters.  (ref: [OC QoS interface counters](https://github.com/openconfig/public/blob/f04dff4e782f9795da6e06323353c44000853a07/release/models/qos/openconfig-qos-interfaces.yang#L325-L335)). Using the proposed #762 model for protection policy and the related OC QoS interface counters would be the most vendor neutral and represent a consistent counter model.  

I'd like to hear from @rolandphung @aredmon8551 @rgwilton @hobbydos if their platforms implementation of control  plane protection can map to [OC QoS interface counters](https://github.com/openconfig/public/blob/f04dff4e782f9795da6e06323353c44000853a07/release/models/qos/openconfig-qos-interfaces.yang#L325-L335),

Reply:
I agree that this is a good pattern to form.

Note, I don't think the references to the "defined sets ACL" model is useful here. This is essentially an optimisation to the existing ACL model to have multiple prefixes that are in reusable sets be referenced. This is _not_ a requirement to implement to have these counters - as long as the input configuration is one-prefix-per-entry then the existing model will continue to operate.

The QoS model being how we get counters - and this applying to the aggregate CPU bound traffic - seems a reasonable way to get the same granularity as is implemented there. This model would be how counters that are more specific than that, and potentially NPU or implementation specific would be exposed.

Reply:
I am ok with having both vendor neutral counters in #762, the vendor neutral integrated circuit aggregate counters here and the vendor specific integrated circuit counters for control plane here.   Regarding the ACL, we can resolve that in #762.

Reply:
@robshakir after the above discussion, shall we mark this comment resolved?   This comment is requesting changes.  Are there changes or open questions you would like answered to see to this PR currently?

Reply:
> In #762 (Control Plane Protection model) the referenced QoS and [defined sets ACL](https://github.com/openconfig/public/blob/master/release/models/defined-sets/openconfig-defined-sets.yang) models have counters. (ref: [OC QoS interface counters](https://github.com/openconfig/public/blob/f04dff4e782f9795da6e06323353c44000853a07/release/models/qos/openconfig-qos-interfaces.yang#L325-L335)). Using the proposed #762 model for protection policy and the related OC QoS interface counters would be the most vendor neutral and represent a consistent counter model.
> 
> I'd like to hear from @rolandphung @aredmon8551 @rgwilton @Hobbydos if their platforms implementation of control plane protection can map to [OC QoS interface counters](https://github.com/openconfig/public/blob/f04dff4e782f9795da6e06323353c44000853a07/release/models/qos/openconfig-qos-interfaces.yang#L325-L335),

@nokia1adam, Could you PTAL?

Reply:
As a standalone PR I'm not sure what this accomplishes. We already have pipeline-counters/host-interface-block, which to me is indistinguishable in scope from the pipeline-counters/control-plane-traffic addition proposed by this PR.

My general issue with the current host-interface-block state model is that it only splits the counters up into "high" and "low" priority buckets, which assumes a scheduling model in which there are two classes (or two queues) and one is strict priority over the other. Real implementations likely have 8, or 16, or 32+ queues from each IC, with the scheduling of those queues possibly WRR instead of strict priority. These queues could also have PIR/CIR rate limits. If you imagined a state-only qos scheduler-policy applied to each IC you would be much closer to reality.

Finally, I don't think this topic should be mixed up with configured CoPP policies (PR 762 for example). The rules that a user configures for protection of their control plane are not 1:1 aligned with the datapath pipeline rules that determine which flows should be extracted to the control plane engine. The former should be presented as CoPP policy statistics while the latter should be presented as pipeline counters.